packages/typespec-go/src/tcgcadapter/clients.ts

```diff
@@ -90,23 +90,32 @@ export class clientAdapter {
         if (param.kind === 'credential') {
           // skip this for now as we don't generate client constructors
           continue;
-        } else if (param.kind === 'endpoint' && param.type.kind === 'endpoint') {
+        } else if (param.kind === 'endpoint') {
+          // for multiple endpoint, we only generate the first one
+          let paramType;
+          if (param.type.kind === 'endpoint') {
+            paramType = param.type;
+          } else {
+            paramType = param.type.values[0];
+          }
           // this will either be a fixed or templated host
           // don't set the fixed host for ARM as it isn't used
           if (this.ta.codeModel.type !== 'azure-arm') {
-            goClient.host = param.type.serverUrl;
+            goClient.host = paramType.serverUrl;
           }
-          if (param.type.templateArguments.length === 0) {
+          if (paramType.templateArguments.length === 0) {
```

Comment:
I've been thinking about this and I'd like to remove the host template and require the full URL just like we do in Rust. See https://github.com/Azure/autorest.rust/blob/main/packages/typespec-rust/src/tcgcadapter/adapter.ts#L392-L395